server/middleware/asyncHandler.js

```diff
@@ -0,0 +1,5 @@
+const asyncHandler = (fn) => (req, res, next) => {
+  Promise.resolve(fn(req, res, next).catch(next));
```

Comment:
what is the point of this function?

Reply:
a usage example 
```
```
export const addNew=asyncHandler(async(req,res,next)=>{
    const newTodo=await Todo.create(req.body);
    if (!newTodo){
       return   next(new Error("failed to add todo to the list"))
    }
    else{
        res.status(200).json({
            success:true,

        })
    }
})
```
```
it is a middleware to better configure the controller function's try/catch blocks
resons to use 
less writing , because no need to open try catch block each time we write a controller function , 
next() is responsible of transporting the error to the catch field in the middleware
very good well knows convention in nodejs  i saw davegray and bradtraverssy and others use it 
express has its own asynchandler package , which include ver similar code , 
in conclusion:
easy to read and understand the controllers  functions